### DISCUSSION TITLE
JLCPCB gerbers not showing correctly in 3D view

### DISCUSSION THREAD

**Discussion by HattoriHanzo031** [strayer/taira-keyboard#8] [Q&A]
I really like this keyboard, it is exactly what I want and I would like to build it, but when I uploaded [taira-1.0-pcb.zip](https://github.com/strayer/taira-keyboard/releases/download/v1.0/taira-1.0-pcb.zip) to JLCPCB, the 3D view is not showing holes or silkscreen for switches. Is this expected, and will they be able to build the board correctly? I never used KiCad for manufacturing at JLCPCB so I'm not sure how to export gerbers properly.

**Comment by HattoriHanzo031** [comment]
Just to report, I bit the bullet and sent it to production anyway and it came back just fine. Assembled it and tested, so far everything works.

**Reply by strayer** [reply]
Hey! So sorry I never answered to your post :( I must have completely missed it. Good to hear everything worked with JLCPCB! Feel free to message me any time if you have feedback or questions!

**Reply by HattoriHanzo031** [reply]
Thanks, I assembled the keyboard and it works flawlessly. Thanks for the great design!
I'm not satisfied with my choice of switches (brown), but that is my fault :)

**Reply by strayer** [reply]
Ah, yes. Same here. When I built the first one I purchased a full set of reds, browns and some variant of clickies (can't remember the color right now). Liked the clickies the most, but I really can't do that in the office and at home if I value my health 😬 The browns felt *very* strange – which surprised me, because I do like MX browns. Settled with the reds and have been using them since. Anyways, I'm happy the keyboard is working out for you! Its still awesome to know that other people are using this in their daily life.

FYI: lowprokb.ca will soon(ish) release [silent linear choc switches](https://lowprokb.ca/collections/switches/products/ambients-silent-choc-switches). I'm **really** excited about those!

FYI2: I'm currently working on a v2.0 of the Taira that will most likely feature a bigger battery (200 mAh) and a display. Not sure how quickly that will be done or if it will be done at all, I have to do some pretty drastic changes to the PCB for that. Top and bottom plates will still be compatible though, so worst case one "only" needs to order new PCBs and the display to upgrade. 🤞

**Reply by HattoriHanzo031** [reply]
Glad to hear you are considering doing the update.

I actually soldered [these](https://ae01.alicdn.com/kf/S679f137322654a1faadfa34bc78fb4548.jpg?width=1000&height=1000&hash=2000) to the controller board and [these](https://ae01.alicdn.com/kf/Sac0a5433f5ee4afa8aab9660d96f571br.jpg?width=1000&height=1000&hash=2000) to the PCB. The upside is that is very easy to solder and plug/unplug, but the downside is that the controller board is few mm higher than normal, but this way there is room for second pouch battery on top of the first one. It probably is not ideal to run two batteries in parallel, but it could potentially double the capacity :)

Those switches look very interesting, but I'm afraid they are a bit outside of my budget (especially with shipping and taxes which are horrible in Serbia). I was thinking of modifying the board to use optical switches and maybe even squeeze in a rotatory encoder, a friend has keyboard with low profile brown optical keychron switches, and they are exactly what I would like switch to feel (and sound) like. I saw reviews of kailh V2 and people usually praise brown V2s, but I never tried them. Bigger battery would definitely come in handy if opticals are used :)

Displays on keyboard looks very nice, but I'm not sure I would use them for anything more than visual effect :)
